packages/graphql/server/resolvers/mutators.ts

```diff
@@ -251,19 +321,14 @@ export const updateMutator = async <TModel extends VulcanDocument>({
   const { typeName } = model.graphql;
   const mutatorName = "update";
   const { schema } = model;
+  let data = { ...dataInput } || modifierToData({ $set: set, $unset: unset });
 
   // get currentUser from context if possible
   if (!currentUser && context.currentUser) {
     currentUser = context.currentUser;
   }
 
-  // OpenCRUD backwards compatibility
-  // TODO: what can we remove?
-  selector = selector || { _id: dataId };
-  let data = { ...dataInput } || modifierToData({ $set: set, $unset: unset });
-
-  // startDebugMutator(collectionName, "Update", { selector, data });
-
+  selector = await selectorOrId(selector, dataId);
```

Comment:
I think I remember the problem: sometimes you may want to use the mutator directly, in some other server code. That's why you may like to pass the selector or the dataId depending on your scenario.

So maybe keep both, forget my previous comment. And you can replace the TODO comment of "selectorOrId" to add an explanation: "allow to use the mutator with either just a document id or a selector".

However, in the mutation (not the mutator, the mutation), you should only use one to keep it simple, probably the selector. DataId would stay only for users that create their own mutation or use the mutator directly.